files/galaxy/dynamic_rules/usegalaxy/total_perspective_vortex/tools.yml

```diff
@@ -533,6 +533,23 @@ tools:
           Too much data, we cannot support such large Trinity assemblies with our
           backend. Please use another server for your job.
 
+  toolshed.g2.bx.psu.edu/repos/iuc/trinity/trinity/2.15.1+galaxy0:
```

Comment:
@cat-bro how can I ensure that this ID works. I have above an ID with `*` and I fear that the wildcard rule matches and not this one? Any idea? 

Please also note that you might want to try out this new version on AU as well. I hope its less memory intense.

Reply:
I believe that both the rules from the * entry and the versioned entry would apply to trinity 2.15.1+galaxy0. If all file sizes are covered in the versioned entry (at the moment file sizes between 20 and 30 are not covered) and the versioned entry is in the file after the wildcard entry, the rules in the versioned entry should override those in the wildcard entry.

Another possibility to change the wildcard key to one that excludes this version, as has been done [here with busco](https://github.com/usegalaxy-au/infrastructure/blob/master/files/galaxy/dynamic_job_rules/production/total_perspective_vortex/tools.yml#L849)

Another possibility would be to merge the two entries into one wildcard rule and use the tool version in the `if`, like [this rule for alphafold](https://github.com/usegalaxy-au/infrastructure/blob/master/files/galaxy/dynamic_job_rules/production/total_perspective_vortex/tools.yml#L652-L655). This would be convoluted to do for many rules, as packaging.version would have to be imported for each if. This really needs helper functions - I'll put in a PR to add helper functions around tool version comparison.

```

Reply:
PR for making tool version comparison easier: https://github.com/galaxyproject/total-perspective-vortex/pull/91/files